File: LuaPythonWorkSessons/WorkSesson6Task5.lua
-- Simple Calculator

function Addition(Number1, Number2)
    local Answer = Number1 + Number2
    print("Your answer is "..Answer)
end

function Subtraction(Number1, Number2)
    local Answer = Number1 - Number2
    print("Your answer is "..Answer)
end

function Division(Number1, Number2)
    local Answer = Number1 / Number2
    print("Your answer is "..Answer)
end

function Multiplication(Number1, Number2)
    local Answer = Number1 * Number2
    print("Your answer is "..Answer)
end

local FirstUserNumber

while type(FirstUserNumber) ~= "number" do
    print("Hello welcome to the calculator please type in your first number input")
    FirstUserNumber = io.read()
    if tonumber(FirstUserNumber) ~= nil then
        print("This works! \n")
        FirstUserNumber = tonumber(FirstUserNumber)
    end
end

local SecondUserInput

while type(SecondUserInput) ~= "number" do
    print("Now type in your second number input")
   SecondUserInput = io.read()
   if tonumber(SecondUserInput) ~= nil then
    print("Alright good job! \n")
    SecondUserInput = tonumber(SecondUserInput)
   end
end

AcceptableOperators = {
    "+",
    "-",
    "/",
    "*"
}

local OperatorWorking = false

local OperatorInput

while OperatorWorking == false do
    print("Now put in a operator of either + - / *")
    OperatorInput = io.read()
    for i = 1, #AcceptableOperators do
        if OperatorInput == AcceptableOperators[i] then
            print("This works")
            OperatorWorking = true
            break
        end
    end
end

if OperatorInput == "+" then
    Addition(FirstUserNumber, SecondUserInput)
elseif OperatorInput == "-" then
    Subtraction(FirstUserNumber, SecondUserInput)
elseif OperatorInput == "/" then
    Division(FirstUserNumber, SecondUserInput)
elseif OperatorInput == "*" then
    Multiplication(FirstUserNumber, SecondUserInput)
end

File: LuaPythonWorkSessons/WorkSesson6Task5.py
def Addition(FirstNumber, SecondNumber):
    print(f"This is your answer for Addition {FirstNumber + SecondNumber}")

def Subtraction(FirstNumber, SecondNumber):
    print(f"This is your answer for Subtraction {FirstNumber - SecondNumber}")

def Division(FirstNumber, SecondNumber):
    print(f"This is your answer for Division {FirstNumber / SecondNumber}")

def Multiplication(FirstNumber, SecondNumber):
    print(f"This is your answer for Multiplication {FirstNumber * SecondNumber}")

AcceptableOperations = [
    "+",
    "-",
    "*",
    "/",
]

FirstUserInput = "Nothing"

while not isinstance(FirstUserInput, (int, float, complex)): # Checks if the FirstUserInput variable is a number value or not while its not a number value it will continue repeating itself until user inputs number value
    print("Type in your first number for the operation")
    FirstUserInput = input()

    if FirstUserInput.isnumeric(): #string based check. Checks if the string contains numbers if it does it converts the string to a number value
        FirstUserInput = int(FirstUserInput)
        print("Good that works!")
    else:
        print("Try again you need a number value")

SecondUserInput = "Nothing"

while not isinstance(SecondUserInput, (int, float, complex)):
    print("Now type in your second number for the operation")
    SecondUserInput = input()
    if SecondUserInput.isnumeric():
        SecondUserInput = int(SecondUserInput)
    else:
        print("Try again you need to type in a number value")

OperationSelection = "Nothing"

while OperationSelection not in AcceptableOperations: # checks if variable OperationSelection has a value equal to string values in AcceptableOperations table loops until OperationSelection has value equal to a string value in the table AcceptableOperation
    print("Now type in your operation type either + - * /")
    OperationSelection = input()
    if OperationSelection == "+":
        Addition(FirstUserInput, SecondUserInput)
    elif OperationSelection == "-":
        Subtraction(FirstUserInput, SecondUserInput)
    elif OperationSelection == "/":
        Division(FirstUserInput, SecondUserInput)
    elif OperationSelection == "*":
        Multiplication(FirstUserInput, SecondUserInput)
    else:
        print("Try again you need to type a single operation exactly how its shown its case sensetive")